### PR TITLE
Memory v3  - msquarednet pull request

### PR DESCRIPTION
**Pull request by msquarednet**
here goes nothing
?

### CODE REVIEW

File: js/cell.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var Cell = function(grid, id){
+	this.grid = grid;
+	this.id = id+=100;			//not strictly required on init;
```
Comment (afeld) [inline js/cell.js:3]: Why are you doing this addition?

File: js/cell.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	//this.status = "HIDDEN";	//nah, just using DOM for now :/
+	//this.selected = false;	//models, views... some other day.
+	this.val;
+	this.init();
+	
+}
+Cell.prototype.init = function() {
+	var chararray = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","X","1","2","3","4","5","6","7","8","9","0"];	//["AA","BB","CC","DD","EE"];	//
+	this.getRandomVal(chararray);
+	while (this.isValAlreadyUsed()) {
```
Comment (afeld) [inline js/cell.js:13]: Minor style note: I would probably have `getRandomVal()` return the value and not actually assign it until a valid one is found.

File: js/grid.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	};
+};
+Grid.prototype.drawGrid = function() {
+	var htmstr = "";
+	var rowstr = "";
+	var cell;
+	for (var i=0; i<this.cellsarray.length; i++) {
+		cell = this.cellsarray[i];
+		cell.id = i;
+		rowstr += '<td id="td_'+cell.id+'" class="unmatched"><span id="cell_'+cell.id+'" class="hidden">' + cell.val + '</span></td>\n'
```
Comment (afeld) [inline js/grid.js:49]: You might consider having each cell create its own markup, which would make this method a bit cleaner.

**Comment by afeld** [comment]
Having the letters always visible makes the game a _lot_ easier :wink:

screen shot 2014-07-05 at 6 29 25 pm

My general feedback is that the `Grid` is doing too much.  You want to push the logic of, say, flipping a tile, into the most specific component that you can, which in that case would be the `Cell`.

Also, while it's accurate to name methods things like `doTDClick()`, try to make the intention more clear by saying something like `onCellClick()`.

Nice work!

File: js/cell.js (hunk 1 above)
Comment (msquarednet) [inline js/cell.js:3]: 'id' was part of debugging. At first I had half the tiles in an array, and tried to make the full array by cloning the first half of it. Of course, the second half of the array was just a reference to the first half. Tracing/messing with 'id's was part of a long effort to realize that I hadn't made the (deep) clone that I had thought that i'd made.

**Pull request closed without merging**